File: app/src/main/java/de/thinksonic/basictemplate/app_initializer/AppInit.java
package de.thinksonic.basictemplate.app_initializer;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.Application;
import android.content.IntentFilter;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.view.WindowManager;

import androidx.annotation.NonNull;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

import de.thinksonic.basictemplate.Interface.ActivityCreated;
import de.thinksonic.basictemplate.Interface.VolleyHandler;
import de.thinksonic.basictemplate.Receivers.NetworkStateReceiver;
import de.thinksonic.basictemplate.Tools.UserPreference;
import de.thinksonic.basictemplate.Tools.Utils;
import de.thinksonic.basictemplate.Tools.VolleyClass;

/*
  Created By : Shesha Vasukhi Prasad
  Date : 15-Jul-2019
  Time : 22:12
  For Project : BasicTemplate
*/


public class AppInit extends Application {

    public static boolean ConntectedToInternet;
    public static boolean SplashCreated = false;
    public VolleyClass serverCall;
    public static RequestQueue requestQueue;
    private NetworkStateReceiver networkStateReceiver;
    @SuppressLint("StaticFieldLeak")
    static Activity mActivity;
    UserPreference userPreference;
    public static StringRequest stringRequest;

    @Override
    public void onCreate() {
        super.onCreate();

        requestQueue = Volley.newRequestQueue(this);
        userPreference = new UserPreference(this);
        networkStateReceiver = new NetworkStateReceiver();

        registerReceiver(networkStateReceiver, new IntentFilter(android.net.ConnectivityManager.CONNECTIVITY_ACTION));

        registerActivityLifecycleCallbacks(new ActivityLifecycleCallbacks() {

            @Override
            public void onActivityCreated(@NonNull final Activity activity,
                                          Bundle savedInstanceState) {

                mActivity = activity;
                activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
                activity.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
                activity.getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);
                serverCall = new VolleyClass(activity,(VolleyHandler) activity);


            }

            @Override
            public void onActivityStarted(@NonNull Activity activity) {
                ((ActivityCreated)activity).initViews();
            }

            @Override
            public void onActivityResumed(@NonNull Activity activity) {
                if(networkStateReceiver!=null) {
                    registerReceiver(networkStateReceiver, new IntentFilter(android.net.ConnectivityManager.CONNECTIVITY_ACTION));
                }else{
                    networkStateReceiver = new NetworkStateReceiver();
                    registerReceiver(networkStateReceiver, new IntentFilter(android.net.ConnectivityManager.CONNECTIVITY_ACTION));
                }
            }

            @Override
            public void onActivityPaused(@NonNull Activity activity) {
                unregisterReceiver(networkStateReceiver);
            }

            @Override
            public void onActivityStopped(@NonNull Activity activity) {

            }

            @Override
            public void onActivitySaveInstanceState(@NonNull Activity activity,@NonNull  Bundle outState) {

            }

            @Override
            public void onActivityDestroyed(@NonNull Activity activity) {

            }
        });

    }

    public static void showNA(){
        ConntectedToInternet = false;
        Utils.ShowNetworkAlert(mActivity);
    }
    public static void dismissNA(){
        ConntectedToInternet = true;
        Utils.DismissNetworkAlert(mActivity);
    }
}

File: app/src/main/java/de/thinksonic/basictemplate/app_initializer/AppInit.kt
package de.thinksonic.basictemplate.app_initializer

import android.app.Activity
import android.app.Application
import android.content.Context
import android.content.IntentFilter
import android.content.pm.ActivityInfo
import android.net.ConnectivityManager
import android.os.Bundle
import android.view.WindowManager
import com.android.volley.RequestQueue
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import de.thinksonic.basictemplate.interfaces.ActivityCreated
import de.thinksonic.basictemplate.interfaces.VolleyHandler
import de.thinksonic.basictemplate.receivers.NetworkStateReceiver
import de.thinksonic.basictemplate.tools.UserPreference
import de.thinksonic.basictemplate.tools.Utils
import de.thinksonic.basictemplate.tools.VolleyClass

/**
 * Created By : Shesha Vasukhi Prasad
 * Date : 15-Jul-2019
 * Time : 22:12
 * For Project : BasicTemplate
*/


class AppInit : Application() {
    lateinit var serverCall: VolleyClass
    private var networkStateReceiver: NetworkStateReceiver? = null
    private var mActivity: Activity? = null
    internal lateinit var userPreference: UserPreference

    override fun onCreate() {
        super.onCreate()

        requestQueue = Volley.newRequestQueue(this)
        userPreference = UserPreference(this)
        networkStateReceiver = NetworkStateReceiver()

        registerReceiver(networkStateReceiver, IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION))

        registerActivityLifecycleCallbacks(object : ActivityLifecycleCallbacks {

            override fun onActivityCreated(activity: Activity,
                                           savedInstanceState: Bundle?) {

                mActivity = activity
                activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
                activity.window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
                activity.window.setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE)
                serverCall = VolleyClass(activity, activity as VolleyHandler)


            }

            override fun onActivityStarted(activity: Activity) {
                (activity as ActivityCreated).initViews()
            }

            override fun onActivityResumed(activity: Activity) {
                if (networkStateReceiver != null) {
                    registerReceiver(networkStateReceiver, IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION))
                } else {
                    networkStateReceiver = NetworkStateReceiver()
                    registerReceiver(networkStateReceiver, IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION))
                }
            }

            override fun onActivityPaused(activity: Activity) {
                unregisterReceiver(networkStateReceiver)
            }

            override fun onActivityStopped(activity: Activity) {

            }

            override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {

            }

            override fun onActivityDestroyed(activity: Activity) {

            }
        })

    }

    companion object {

        var ConntectedToInternet: Boolean = false
        var SplashCreated = false
        lateinit var requestQueue: RequestQueue
        var stringRequest: StringRequest? = null

        fun showNA(context: Context) {
            ConntectedToInternet = false
            Utils.ShowNetworkAlert(context as Activity)
        }

        fun dismissNA(context: Context) {
            ConntectedToInternet = true
            Utils.DismissNetworkAlert(context as Activity)
        }
    }
}
